crates/kitsune_p2p/dht/src/region_set/ltcs.rs

```diff
@@ -0,0 +1,194 @@
+use once_cell::sync::OnceCell;
+
+use crate::{
+    arq::*,
+    error::{GossipError, GossipResult},
+    op::OpRegion,
+    spacetime::*,
+};
+use derivative::Derivative;
+
+use super::{Region, RegionCoords, RegionData, RegionDataConstraints};
+
+/// A compact representation of a set of [`RegionCoords`].
+/// The [`TelescopingTimes`] generates all relevant [`TimeSegment`]s, and the
+/// [`SpaceSegment`]s are implied by the [`ArqBoundsSet`].
+///
+/// LTCS stands for Logarithmic Time, Constant Space.
+#[derive(Debug, PartialEq, Eq, derive_more::Constructor, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "test_utils", derive(Clone))]
+pub struct RegionCoordSetLtcs {
+    pub(super) times: TelescopingTimes,
+    pub(super) arq_set: ArqBoundsSet,
+}
+
+impl RegionCoordSetLtcs {
+    /// Generate the LTCS region coords given the generating parameters.
+    /// Each RegionCoords is paired with the relative spacetime coords, which
+    /// can be used to pair the generated coords with stored data.
+    pub fn region_coords_flat(&self) -> impl Iterator<Item = ((u32, u32), RegionCoords)> + '_ {
```

Comment:
the `+ '_` is implied, isn't it? can you drop that?

```suggestion
    pub fn region_coords_flat(&self) -> impl Iterator<Item = ((u32, u32), RegionCoords)> {
```

Reply:
I was getting errors about this, but just tried it in the rust playground and it worked, so I will try again